### DISCUSSION TITLE
Respect "Select a container for each new tab" setting when opening tab with CTRL + T shortcut

### DISCUSSION THREAD

**Discussion by mr-cheffy** [zen-browser/desktop#847] [Ideas]
### What feature would you like to see?

Self-explanatory title

### What platform are you seeing the problem on?

Windows

 
 ~ _Created automatically from issue: https://github.com/zen-browser/desktop/issues/407_

**Comment by rubenmate** [comment]
I'm resurrecting this. What is the best way to open a new tab and select a specific container? I require to do this a lot and I can't find a way to do it that doesn't require a lot of steps.

**Comment by KittyKatt** [comment]
I can confirm that this feature in Containers doesn't seem to work whatsoever in Zen. I do have the MAC extension installed and have the radio button checked for "Select a container for each new tab" in the bulit-in Containers settings. However, every tab I open with Ctrl+T never prompts me to select a container. Worth noting that the new tab "+" toolbar button exhibits the same behavior. The only way to select a tab for every new container is to us the "New Tab" button and use the context menu that appears to select a container OR to assign specific containers to keybinds, which isn't super great when the functionality is right there built-in to support the regular tab open keybind.

As an additional note, I'm running this on KDE in a Linux environment. Using Zen Browser 1.17.10b (Firefox 145.0.2).

**Comment by SimJoSt** [comment]
It would be great, if this would be configurable.
Right now, having the option to open tabs without container selection through the keyboard shortcut, works pretty well.
In the future, I might want to change it. So an option to toggle when the container selection should appear would be better, than forcing a default on everyone.

As this feature in Zen comes from the Multi-Account Container Firefox add-on, it might make sense to request that feature in it's GitHub repository: https://github.com/mozilla/multi-account-containers
There are already 2 issues opens regarding this:
- https://github.com/mozilla/multi-account-containers/issues/2535
- https://github.com/mozilla/multi-account-containers/issues/462

However, in the former there is some mention about some work that needs to be done in Firefox itself, to allow changing the new tab behaviour from the add-on: https://github.com/mozilla/multi-account-containers/issues/2535#issuecomment-1749788798
As tab containers are a native feature to Zen, it might be possible/easier to make the necessary adjustments to Zen, than to Firefox and the add-on.

**Comment by xandemon** [comment]
I so want this feature. But considering this issue is from Aug 26, 2024 and there is no meaningful response, I should consider this a dead end??

**Reply by KittyKatt** [reply]
I ultimately ended up switching away from Zen because a large amount of my workflow depends on tabs opening in specific containers or being about to select an arbitrary container for a tab when I open it. For a while, I was just reopening tabs in the right container or moving them to different workspaces, but that became too frustrating.

I think part of the problem with this specific behavior and Zen Is that I don't think Zen actually technically opens up a "new tab" tab when you open a new tab? I remember looking into this a while back and coming to the same conclusion that I just don't think this is (or will be any time soon) a priority for the team (which is fine, by the way).

**Reply by xandemon** [reply]
Oh, very strong of you to make the switch really. Not sure I can do that (at least not anytime soon). For now, I'm making do with the Firefox extension @SimJoSt linked above but not a very convenient/native way to do this and some important things are still missing there but that's the best we have right now I guess.

Anyway, I'd love to know what you switched to though. I don't see many good options tbh. Arc? Or Brave maybe??
